lib/commands/empty_bucket.go

```diff
@@ -0,0 +1,98 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GetTerminus/s3helper/lib/parser"
+	"github.com/GetTerminus/s3helper/lib/provider"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/pkg/errors"
+)
+
+// EmptyBucketCommand represents the options that can be passed to the empty-bucket subcommand.
+type EmptyBucketCommand struct {
+	Bucket string `short:"b" long:"bucket" value-name:"bucket" description:"the bucket to empty" required:"true"`
+}
+
+func init() {
+	var cmd EmptyBucketCommand
+	parser.OptParser.AddCommand(
+		"empty-bucket",
+		"Erase all contents in an s3 bucket",
+		"Erase all objects and delete markers in an s3 bucket",
+		&cmd,
+	)
+}
+
+// Execute implements the interface for the go-flags subcommand.
+func (cmd *EmptyBucketCommand) Execute(args []string) error {
+	fmt.Fprintf(os.Stdout, "Deleting contents of s3://%s\n", cmd.Bucket)
+
+	// Create a session and get an s3 handle
+	session, err := provider.AWSSession()
+	if err != nil {
+		return errors.Wrap(err, "error getting aws session")
+	}
+
+	svc := s3.New(session)
+
+	totalDeleted := 0
+
+	for {
+		// 2 API calls are needed to delete things in s3, the first will get a list
+		// of the objects in the bucket, and the second to delete those things
+		resp, err := svc.ListObjectVersions(&s3.ListObjectVersionsInput{
+			Bucket: aws.String(cmd.Bucket),
+		})
+		if err != nil {
+			return errors.Wrap(err, "error listing object versions on bucket")
```

Comment:
whats gained with this errors.Wrap? I have seen it used before but have not seen the benefit yet

Reply:
It's useful for building a chain of events that lead back to the root cause of the error. Each message that was previously wrapped is preserved.